src/core/datastore/processors/cache-offline-data-processor.js

```diff
@@ -111,16 +113,12 @@ export class CacheOfflineDataProcessor extends OfflineDataProcessor {
   }
 
   _processUpdate(collection, data, options) {
-    let offlineEntity;
     return super._processUpdate(collection, data, options)
-      .then((result) => {
-        offlineEntity = result;
-        return this._networkRepository.update(collection, data, options);
-      })
+      .then(() => this._networkRepository.update(collection, data, options))
```

Comment:
Is this updating on the network first, followed by locally? If network fails, is the change persisted locally? (it should be) 
How does it compare to current behavior?

Reply:
It updates offline first, then over the network. The offline update is right above - the `super._processUpdate()` part. That's the `_processUpdate()` of `OfflineDataProcessor`. I believe it's the same as the current behaviour.